### PR TITLE
63: Edit account

### LINKED ISSUE #60: Manage Account
**Description:**

Students will be able to create an account, login into the app with their credentials, and manage their account information.

**Related User Stories:**

- [x] #62 As a student, I want to create an account so that I can use the BisonCoin app.
- [x] #55 As a student, I want to manage my account so that I can update my account information.
- [x] #125 As a student, I want the ability to login so that I can access my account.
- [x] #126 As a student, I want the ability to log out so that I can leave my account.

### LINKED ISSUE #63: Implement edit account functionality
**Description:**

Implement functionality to edit user information.

**Acceptance Criteria:**
- [x] Toggle edit functionality on button click
- [x] Save new data through calling service endpoints on save button click

### PR DESCRIPTION
**Pull request by ZackHolmberg**
closes #63

### CODE REVIEW

File: src/frontend/src/pages/LoginPage.vue [hunk 1]
```diff
@@ -42,6 +42,14 @@ import Button from "../components/Button.vue";
     TextInput,
     Button,
   },
+  methods: {
```
Comment (madisonfines) [inline src/frontend/src/pages/LoginPage.vue]: I think this got switched back when merged with staging, please move into export default class!

File: src/frontend/src/store/index.ts [hunk 2]
```diff
@@ -108,8 +112,64 @@ export default new Vuex.Store({
...
+          "new_password": password
+        })
+        .then(
+          () => {
+            commit("MUTATION_SET_WALLET_CREATED", true);
+            commit("MUTATATION_SET_LOADING", false);
+            commit("MUTATATION_SET_FIRST_NAME", firstName);
+            commit("MUTATATION_SET_LAST_NAME", lastName);
+            commit("MUTATATION_SET_PASSWORD", password);
+
```
Comment (madisonfines) [inline src/frontend/src/store/index.ts]: Remove these spaces!

File: src/frontend/src/store/index.ts [hunk 3]
```diff
@@ -108,8 +112,64 @@ export default new Vuex.Store({
...
+
+          },
+          (err) => {
+            commit("MUTATATION_SET_LOADING", false);
+            Vue.$toast.error(
+              err.response.data.error
+                ? err.response.data.error
+                : "An error occurred. Please try again.",
```
Comment (madisonfines) [inline src/frontend/src/store/index.ts]: I think we can create a constant for this string since we use it a lot for errors!

**Review by madisonfines — CHANGES_REQUESTED** [review]

**Review by abhisachdev17 — APPROVED** [review]
The approval shall be with you

**Review by madisonfines — APPROVED** [review]
office

**Pull request merged**